### DISCUSSION TITLE
Export tokens in a single file

### DISCUSSION THREAD

**Discussion by camposcristian** [mikaelvesavuori/figmagic#166] [Ideas]
Instead of `/tokens/colors.ts` , `/tokens/spacing.ts`, `/tokens/fontFamilies.ts`, etc, is there any way to get all tokens into one single file `tokens.ts`? This will help following conventions for other libraries like Radix or Chakra UI

**Comment by mikaelvesavuori** [comment]
Hey @camposcristian!

This will sound really dumb but I never got informed something happened in the GitHub Discussions... Sorry about that!

I've done a quick investigation into this and it should be possible - at least more or less. The issues making this harder are:
* There will need to be some modifications to parts of the code outputting the file - totally possible and I got the format to look about right. This is the easy part, I already see we can get this to work.
* The issue is that there is currently no option for choosing this single file output, so it will need to be created, tested and all of that, plus ensuring it can be passed to the correct part of the code. This is doable but messy and I can't prioritize it right now.
* Single file output breaks the current implementation of element generation. If you choose not to generate React elements it will work, but we need to do the above work in the use case code for React element generation too, so we can load only from a single file.
